src/pages/index.js

```diff
@@ -1,106 +1,204 @@
+/* -------------------------------------------------------------------------- */
 
-editProfilePopup.setEventListeners();
+let userId;
+let cardList;
+
+Promise.all([Api.getInitialCards(), Api.getUserInfo()])
+.then(([initialCards, userData]) => {
+  userId = userData._id;
+  userDataInfo.setUserInfo(userData.name, userData.about);
+  userInfo.setProfileAvatar(userData.avatar);
+  cardList = new Section(
+    {
+      items: initialCards,
+      renderer: (data) => {
+        const newCard = createCard(data);
+        cardList.addItem(newCard);
+      },
+    },
+    cardList
+  );
+  cardList.renderItems();
+})
+.catch(() => console.log(err));
 
 /* -------------------------------------------------------------------------- */
-/*                                Section Class                               */
+/*                              Profile Functions                             */
 /* -------------------------------------------------------------------------- */
 
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: ({ name, link }) => {
-      const newCard = createCard({ name, link });
-      cardListSection.addItem(newCard);
-    },
-  },
-  cardListEl
-);
+function openProfilePopup() {
+  const { nameSelector, descriptionSelector} = userInfo.getUserInfo();
+  profileTitleEl.value = nameSelector;
+  profileDescriptionEl.value = descriptionSelector;
+  editProfileFormValidator.toggleButtonState();
+  editProfilePopup.open();
+}
 
-cardListSection.renderItems();
+function handleProfileFormSubmit({ name, about }) {
+  editProfilePopup.setLoading(true);
+  Api
+    .updateUserInfo(name, about)
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editProfilePopup.setLoading(false, "Save");
+    });
+}
 
 /* -------------------------------------------------------------------------- */
-/*                                PopupWithForm                               */
+/*                           Profile Event Listeners                          */
 /* -------------------------------------------------------------------------- */
 
-const newCardPopup = new PopupWithForm(profileAddModal, submitCard);
-const previewImagePopup = new PopupWithImage(previewImageModal);
-previewImagePopup.setEventListeners();
+profileEditButton.addEventListener('click', openProfilePopup);
 
-profileAddBtn.addEventListener('click', () => {
-  addFormValidator.toggleButtonState();
-  newCardPopup.open();
+profileAvatarIcon.addEventListener('click', () => {
+  avatarFromValidator.toggleButtonState();
+  editAvatarPopup.open();
 })
 
- newCardPopup.setEventListeners();
+function handleProfileAvatarSubmit(url) {
+  editAvatarPopup.setLoading(true);
+  Api
+   .setUsereAvatar(url)
+   .then((userData) => {
+    userInfo.setProfileAvatar(userData.avatar);
```

Comment:
It's important to format code correctly, use the indentation hierarchy, and split your code. Please check these sources to study the standard: [https://code.s3.yandex.net/web-developer/landings/design-rules-en/index.html](https://code.s3.yandex.net/web-developer/landings/design-rules-en/index.html), [https://javascript.info/coding-style](https://javascript.info/coding-style).

To make sure your code is formatted correctly, we suggest using Prettier (https://prettier.io/). It can be integrated with any code editor or IDE.